packages/cds-odata-annotation-converter/test/data/parser/flattened-nested-record/assignment.txt

```diff
@@ -0,0 +1,4 @@
+UI : {
+    HeaderInfo.Title.$Type : 'UI.DataFieldForAnnotation',
+    HeaderInfo.Title.Target : '@Communication.Contact'
+}
```

Comment:
We should have a bit more test cases here:
1. where we have also the $Type directly on annotation (valid)
2. where it does not have a string value (invalid)
3. where $Type is in the middle of a flattened structure e.g `HeaderInfo.$Type.Title` (invalid)

Maybe you can think of some other scenarios which could have issues.

Reply:
added